climakitae/core/data_interface.py

```diff
@@ -1577,3 +1586,426 @@ def _display_select(self):
+    """Get the variable name corresponding to its ID, downscaling method, and timescale
+    Enables the _get_user_friendly_catalog function to get the name of a variable corresponding to a set of user inputs
+    i.e we have several different precip variables, corresponding to different downscaling methods (WRF vs. LOCA)
+
+    Parameters
+    ----------
+    variable_id: str
+    downscaling_method: str
+    timescale: str
+    var_pd: pd.DataFrame
+        Variable descriptions table
+
+    Returns
+    -------
+    var_name: str
+        Display name of variable from variable descriptions table
+        Will match what the user would see in the selections GUI
+    """
+    # Query the table based on input values
+    var_pd_query = var_pd[
+        (var_pd["variable_id"] == variable_id)
+        & (var_pd["downscaling_method"] == downscaling_method)
+    ]
+
+    # Timescale in table needs to be handled differently
+    # This is because the monthly variables are derived from daily variables, so they are listed in the table as "daily, monthly"
+    # Hourly variables may be different
+    # Querying the data needs special handling due to the layout of the csv file
+    var_pd_query = var_pd_query[var_pd_query["timescale"].str.contains(timescale)]
+
+    # This might return nothing if the variable is one we don't want to show the users
+    # If so, set the var_name to nan
+    # The row will later be dropped
+    if len(var_pd_query) == 0:
+        var_name = "NONE"
+
+    # If a variable name is found, grab and return its proper name
+    else:
+        var_name = var_pd_query["display_name"].item()
+
+    return var_name
+
+
+def get_data_options(
+    variable=None,
+    downscaling_method=None,
+    resolution=None,
+    timescale=None,
+    scenario=None,
+    tidy=True,
+):
+    """Get data options, in the same format as the Select GUI, given a set of possible inputs.
+    Allows the user to access the data using the same language as the GUI, bypassing the sometimes unintuitive naming in the catalog.
+    If no function inputs are provided, the function returns the entire AE catalog that is available via the Select GUI
+
+    Parameters
+    ----------
+    variable: str, optional
+        Default to None
+    downscaling_method: str, optional
+        Default to None
+    resolution: str, optional
+        Default to None
+    timescale: str, optional
+        Default to None
+    scenario: str, optional
+        Default to None
+    tidy: boolean, optional
+        Format the pandas dataframe? This creates a DataFrame with a MultiIndex that makes it easier to parse the options.
+        Default to True
+
+    Returns
+    -------
+    cat_subset: pd.DataFrame
+        Catalog options for user-provided inputs
+
+    """
+
+    # Get intake catalog and variable descriptions from DataInterface object
+    data_interface = DataInterface()
+    var_pd = data_interface.variable_descriptions
```

Comment:
Does `pd` in `var_pd` refer to pandas? `df` would be more intuitive for me. Or `variable_descriptions` like how the table is named elsewhere.

Reply:
Yes, it does! I changed it now, good suggestion :D